src/register-evm-agent/src/register_evm_agent/src/reservation.rs

```diff
@@ -103,4 +135,27 @@ impl<'a> ReservationService<'a> {
 
         Ok(())
     }
+
+    /// Wait for the transaction to be finalized
+    async fn wait_for_transaction(&self, tx_hash: H256) -> Result<()> {
+        const MAX_RETRIES: u32 = 10;
+
+        for _ in 0..MAX_RETRIES {
+            info!("waiting for transaction to be finalized...");
+            tokio::time::sleep(std::time::Duration::from_secs(1)).await;
+
+            let tx_receipt = self
+                .client
+                .eth_get_transaction_receipt(tx_hash.clone())
+                .await??;
+
+            if let Some(receipt) = tx_receipt {
+                if receipt.status == Some(1_u64.into()) {
```

Comment:
I think that in opposite case we should return error immediately, there is no point to continue the cycle